tools/checkcomments/checkcomments.go

```diff
@@ -59,3 +70,71 @@ func run(pass *analysis.Pass) (any, error) {
 
 	return nil, nil
 }
+
+// isIssueopen check the issue open or closed.
+func isIssueOpen(todoText string) bool {
+	issueURL := getURL(todoText)
+	if issueURL == "" {
+		return false
+	}
+
+	token := ""
+	ctx := context.Background()
+	httpClient := oauth2.NewClient(ctx, oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	))
+
+	client := github.NewClient(httpClient)
+
+	owner := "owner"
+	repo := "repos"
+	issueNumber, err := getIssueNumber(issueURL)
+	if err != nil {
+		log.Fatalf("error in getting issue number: %s", err.Error())
+		return false
+	}
+	issue, _, err := client.Issues.Get(ctx, owner, repo, issueNumber)
+	if err != nil {
+		log.Fatalf("error in getting status of issue: %s for issue: %s", err.Error(), issueURL)
+		return false
+	}
+
+	return issue.GetState() == "open"
+}
+
+// extracting url from TODO comment if present.
+func getURL(todoText string) string {
+	arrText := strings.Split(todoText, " ")
+	for _, text := range arrText {
+		if strings.Contains(text, "https://") {
+			parts := strings.Split(text, "/")
+			if len(parts) >= 5 {
+				issueNumber := parts[6]
+				apiURL := fmt.Sprintf("https://api.github.com/repos/FerretDB/FerretDB/issues/%s", issueNumber)
```

Comment:
What are we doing there? We create a URL to check issue status that we don't actually use. Why?

Reply:
Yes correct we don't want that as creating url as we are using  github client to get the state value we can remove it!!